src/SFML/Window/DRM/DRMContext.hpp

```diff
@@ -0,0 +1,196 @@
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Enable or disable vertical synchronization
+    ///
+    /// Activating vertical synchronization will limit the number
+    /// of frames displayed to the refresh rate of the monitor.
+    /// This can avoid some visual artifacts, and limit the framerate
+    /// to a good value (but not constant across different computers).
+    ///
+    /// \param enabled: True to enable v-sync, false to deactivate
+    ///
+    ////////////////////////////////////////////////////////////
+    virtual void setVerticalSyncEnabled(bool enabled);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Create the context
+    ///
+    /// \param shared       Context to share the new one with (can be NULL)
+    /// \param bitsPerPixel Pixel depth, in bits per pixel
+    /// \param settings     Creation parameters
+    ///
+    ////////////////////////////////////////////////////////////
+    void createContext(DRMContext* shared);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Create the surface
+    ////////////////////////////////////////////////////////////
+    void createSurface(unsigned int, unsigned int, unsigned int bpp, bool scanout);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Destroy the EGL surface
+    ///
+    /// This function must be called when the activity is stopped, or
+    /// when the orientation change.
+    ///
+    ////////////////////////////////////////////////////////////
+    void destroySurface();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get the best EGL visual for a given set of video settings
+    ///
+    /// \param display      EGL display
+    /// \param bitsPerPixel Pixel depth, in bits per pixel
+    /// \param settings     Requested context settings
+    ///
+    /// \return The best EGL config
+    ///
+    ////////////////////////////////////////////////////////////
+    static EGLConfig getBestConfig(EGLDisplay display, unsigned int bitsPerPixel, const ContextSettings& settings);
+
+    static GlFunctionPointer getFunction(const char* name);
+
+    static struct drm *get_drm();
+
+private:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Helper to copy the picked EGL configuration
+    ////////////////////////////////////////////////////////////
+    void updateSettings();
+
+    ////////////////////////////////////////////////////////////
+    // Member data
+    ////////////////////////////////////////////////////////////
+    EGLDisplay  m_display; ///< The internal EGL display
+    EGLContext  m_context; ///< The internal EGL context
+    EGLSurface  m_surface; ///< The internal EGL surface
+    EGLConfig   m_config;  ///< The internal EGL config
+
+    struct gbm_bo *m_cur_bo;
+    struct gbm_bo *m_next_bo;
+    struct gbm_surface *m_gbm_surface;
```

Comment:
I believe we don't need the keyword `struct` here, or in other similar places.

Reply:
Except when declaring or defining a type, `struct` can be omitted.

Reply:
Omitted.